### PR TITLE
ninja spacecreep

### PR DESCRIPTION
**Pull request by SonixApache**
+ Helmet is pressurized for 2 minutes. Time is running out!
+ Shoes are magnetized for 2.2 minutes. Hurry up!
+ Suit is pressurized for 2.5 minutes. Panic!

also adds an override to the `togglemagpulse()` proc so shoes can call it on themselves and stuff.

🆑 
 - tweak: Space ninjas now get about two minutes of space protection.

### CODE REVIEW

File: code/datums/gamemode/role/ninja.dm [hunk 1]
```diff
@@ -598,11 +600,22 @@ Suit and assorted
...
 /obj/item/clothing/shoes/ninja
 	name = "ninja shoes"
-	desc = "A pair of running shoes, excellent for running and even better for smashing skulls."
+	desc = "A pair of ninja shoes, excellent for running and even better for smashing skulls."
 	icon_state = "s-ninja"
 	item_state = "s-ninja"
 	permeability_coefficient = 0.01
-	//flags = NOSLIP too powerful
+	mag_slow = NO_SLOWDOWN
+	clothing_flags = NOSLIP | MAGPULSE
```
Comment (ShiftyRail) [inline code/datums/gamemode/role/ninja.dm:608]: `clothing_flags = NOSLIP`
:thinking:

Comment (SonixApache) [reply code/datums/gamemode/role/ninja.dm:608]: Magboots are `NOSLIP | MAGPULSE` when active, `togglemagpulse` toggles both flags.
This means the pair of apprentice shoes ends up with neither.
https://github.com/vgstation-coders/vgstation13/blob/b82d367101190d0d114b59f8a2348652e779aa8f/code/modules/clothing/clothing.dm#L481-L491

**Comment by Coxswain-Navigator** [comment]
ok but do they actually get thrown at the station properly now?

**Comment by SonixApache** [comment]
that was fixed in #23002

**Pull request merged**